src/equations/numerical_fluxes.jl

```diff
@@ -318,4 +318,28 @@ end
 end
 
 
+"""
+    FluxUpwind(splitting)
+
+A numerical flux `f(u_left, u_right) = f⁺(u_left) + f⁻(u_right)` based on
+flux vector splitting.
+
+The [`SurfaceIntegralUpwind`](@ref) with a given `splitting` is analytically
+equivalent to the [`SurfaceIntegralStrongForm`](@ref) with `FluxUpwind(splitting)`
+as numerical flux.
+"""
+struct FluxUpwind{Splitting}
+  splitting::Splitting
+end
+
+@inline function (numflux::FluxUpwind)(u_ll, u_rr, orientation::Int, equations)
+  @unpack splitting = numflux
+  fm = splitting(u_rr, Val{:minus}(), orientation, equations)
+  fp = splitting(u_ll, Val{:plus}(),  orientation, equations)
```

Comment:
Because there is basically nothing to be gained (in terms of performance) and the current version is more explicit. In a combined call, it may not be clear whether `f_minus = f_minus(u_ll)`, `f_minus = f_minus(u_rr)`, or something depending on both `u_ll` and `u_rr`.

Reply:
No, makes absolutely sense as you did it! /c